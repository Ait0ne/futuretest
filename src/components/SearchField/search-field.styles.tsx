import styled from 'styled-components';


export const SeacrFieldContainer = styled.div`
display: flex;
flex-direction: row;
justify-content: flex-end;
align-items: center;
>div>svg {
    position: absolute;
    right: 17px;
    font-size: 20px;
    top: calc(50% - 10px);
    cursor: pointer;
    color: rgba(0,0,0,0.3)
}
`



