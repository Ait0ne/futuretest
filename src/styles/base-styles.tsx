import styled from 'styled-components';

interface ButtonProps {
    active?: boolean
}

export const Button = styled.button<ButtonProps>`
cursor: pointer;
min-width: 64px;
padding: 6px 16px;
color: rgba(0, 0, 0, 0.87);
text-transform: uppercase;
font-weight: 500;
border-radius:4px;
border: none;
margin: 5px;
background-color: #e0e0e0;
box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2), 0px 2px 2px 0px rgba(0,0,0,0.14), 0px 1px 5px 0px rgba(0,0,0,0.12);
&:disabled {
    color: rgba(0, 0, 0, 0.26)
}
`

export const Input = styled.input`
padding: 12px 25px 12px 14px;
border-radius: 4px;
width:25ch;
border: 1px solid rgba(0,0,0,0.3);
font-size: 1rem;
margin: 5px 10px;
&:focus {
    outline:none;
}

`