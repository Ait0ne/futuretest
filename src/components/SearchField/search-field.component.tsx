import React, { useState } from 'react';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faTimes} from '@fortawesome/free-solid-svg-icons';

import {SeacrFieldContainer} from './search-field.styles';
import {Button, Input}from '../../styles/base-styles';

interface SearchFieldProps {
    changeFilterValue: (text:string)=> void
}


const SearchField:React.FC<SearchFieldProps> = ({changeFilterValue}) => {
    const [filter, setFilter] = useState('')
    
    const handleInputChange = (event:React.ChangeEvent<HTMLInputElement>) => {
        setFilter(event.target.value)
    }

    const clearFilter = () => {
        setFilter('')
        changeFilterValue('')
    }

    const handleEnterPress = (event:React.KeyboardEvent<HTMLInputElement>) => {
        if (event.key==='Enter') {
            changeFilterValue(filter)
        }
    }

    return (
        <SeacrFieldContainer>
            <div style={{position:'relative'}}>
                <Input 
                type='text' 
                name='filter' 
                value={filter} 
                onChange={handleInputChange} 
                onKeyPress={handleEnterPress}
                placeholder='Поиск'
                />
                <FontAwesomeIcon onClick={clearFilter} icon={faTimes}/>
            </div>
            <Button onClick={() => changeFilterValue(filter)}>Найти</Button>
        </SeacrFieldContainer>
    )
}

export default SearchField;