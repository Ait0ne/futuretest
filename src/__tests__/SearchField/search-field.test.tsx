import React from 'react';
import {shallow, mount} from 'enzyme';
import SearchField from '../../components/SearchField/search-field.component';
import {Input, Button} from '../../styles/base-styles';

describe('search field', () => {
    const changeFilterValue= jest.fn()

    const searchField = mount(<SearchField changeFilterValue={changeFilterValue}/>)
    
    it('renders correctly', () => {
        expect(searchField).toMatchSnapshot()
    })

    describe('input field', () => {
        const input:any = searchField.find('input').at(0)
        afterEach(() => {
            jest.clearAllMocks()
        })

        it('renders one input', () => {
            expect(input.length).toEqual(1)
        })

        it('is empty', () => {
            expect(input.prop('value')).toEqual('')
        })

        it('calls changeFilterValue function with correct values on enter Press', () => {
            input.instance().value = 'hello'
            input.simulate('change')
            input.simulate('keypress', {key: 'Enter'})

            expect(changeFilterValue).toHaveBeenCalledWith('hello')
        })

        it("doesn't call changeFilterValue on keypress other then enter", () => {
            input.simulate('keypress', {key: 'space'})

            expect(changeFilterValue).toHaveBeenCalledTimes(0)
        })

    })

    describe('search button', () => {
        const button = searchField.find('button').at(0)
        const input = searchField.find('input').at(0)

        it('renders one button', () => {
            expect(button.length).toEqual(1)
        })

        it('calls changeFilterValue function with correct values', () => {
            input.simulate('change', { target: { name: 'filter', value: 'andy' }})
            button.simulate('click')
            expect(changeFilterValue).toHaveBeenCalledWith('andy')
        })
    })

    describe('clear button', () => {
        const clearButton = searchField.find('FontAwesomeIcon')

        it('renders one clearFilter button', () => {
            expect(clearButton.length).toEqual(1)
        })

        it('calls changeFilterValue with correct value on click', () => {
            clearButton.simulate('click')
            expect(changeFilterValue).toHaveBeenCalledWith('')
        })

    })

})