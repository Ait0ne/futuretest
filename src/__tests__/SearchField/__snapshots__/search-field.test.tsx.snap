// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search field renders correctly 1`] = `
<SearchField
  changeFilterValue={[MockFunction]}
>
  <styled.div>
    <div
      className="sc-bdnylx dcymWg"
    >
      <div
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <styled.input
          name="filter"
          onChange={[Function]}
          onKeyPress={[Function]}
          placeholder="Поиск"
          type="text"
          value=""
        >
          <input
            className="sc-dlnjPT lovDsq"
            name="filter"
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="Поиск"
            type="text"
            value=""
          />
        </styled.input>
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                352,
                512,
                Array [],
                "f00d",
                "M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z",
              ],
              "iconName": "times",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          onClick={[Function]}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-times fa-w-11 "
            data-icon="times"
            data-prefix="fas"
            focusable="false"
            onClick={[Function]}
            role="img"
            style={Object {}}
            viewBox="0 0 352 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </FontAwesomeIcon>
      </div>
      <styled.button
        onClick={[Function]}
      >
        <button
          className="sc-gtssRu eOovzA"
          onClick={[Function]}
        >
          Найти
        </button>
      </styled.button>
    </div>
  </styled.div>
</SearchField>
`;
